src/mlpack/core/optimizers/smorms3/smorms3.hpp

```diff
@@ -0,0 +1,149 @@
+/**
+ * @file smorms3.hpp
+ * @author Vivek Pal
+ *
+ * SMORMS3 i.e. squared mean over root mean squared cubed optimizer. It is a
+ * hybrid of RMSprop, which estimates a safe and optimal distance based on
+ * curvature and Yann LeCun’s method in "No more pesky learning rates".
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SMORMS3_SMORMS3_HPP
+#define MLPACK_CORE_OPTIMIZERS_SMORMS3_SMORMS3_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/optimizers/sgd/sgd.hpp>
+
+#include "smorms3_update.hpp"
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * SMORMS3 is an optimizer that estimates a safe and optimal distance based on
+ * curvature and normalizing the stepsize in the parameter space. It is a hybrid
+ * of RMSprop and Yann LeCun’s method in "No more pesky learning rates".
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @misc{Funk2015,
+ *   author    = {Simon Funk},
+ *   title     = {RMSprop loses to SMORMS3 - Beware the Epsilon!},
+ *   year      = {2015}
+ * }
+ * @endcode
```

Comment:
If you want to use the BiBTeX citation here we should probably add the URL: http://sifter.org/~simon/journal/20150420.html

Alternately you could just cite the URL and we can hope Simon never takes his website down. :)